pkg/kubelet/kubelet.go

```diff
@@ -2176,8 +2200,23 @@ func (kl *Kubelet) StreamingConnectionIdleTimeout() time.Duration {
 	return kl.streamingConnectionIdleTimeout
 }
 
+func (kl *Kubelet) getStaticPodUID(uid types.UID) (types.UID, bool) {
+	kl.podLock.RLock()
+	defer kl.podLock.RUnlock()
+	return kl.mirrorPods.GetStaticUID(uid)
+}
+
 // GetContainerInfo returns stats (from Cadvisor) for a container.
 func (kl *Kubelet) GetContainerInfo(podFullName string, uid types.UID, containerName string, req *cadvisorApi.ContainerInfoRequest) (*cadvisorApi.ContainerInfo, error) {
+
+	// If the UID belongs to a mirror pod, translate it to the UID of its
+	// static counterpart.
+	if uid != "" {
```

Comment:
WDYT of doing this logic in getStaticPodUID() since we do the same thing in all callsites? Maybe getPodUID()